exporter/node_stats.py

```diff
@@ -6,86 +6,119 @@
 
 logger = logging.getLogger(__name__)
 
+FREEPOOL_AGGREGATE_ID = 1
+LABELS = [
+    'region',
+    'name',
+    'node_id',
+    'maintenance',
+    'provision_state',
+    'node_type',
+    'gpu',
+    'project_name',
+    'reserved']
+
+
 class NodeStats(OSBase):
     """Class to report the statistics on OpenStack Nodes"""
 
     def __init__(self, oscache, osclient):
         super(NodeStats, self).__init__(oscache, osclient)
 
     def build_cache_data(self):
+        """Return list of stats to cache."""
+        nodes = self.get_nodes()
+        hosts = self.get_hosts_by_node_id()
+        reservations = self.get_reservations_by_node_id()
+
+        for node in nodes:
+            node.node_type = hosts[node.uuid]['node_type']
+            node.gpu = hosts[node.uuid]['gpu']
+            node.project_name = reservations.get(node.uuid, None)
+            node.reserved = node.project_name is not None
```

Comment:
I don't think we really need an explicit "reserved" label, given it is computed as the result of another label. It is easy enough in Prometheus to specify a query like:

```
openstack_node_totals{project_name!="None"}
```